modules/apps.nix

```diff
@@ -82,12 +82,13 @@ in
   };
   config = {
     flake.apps =
-      mapAttrs
-        (k: v: v.apps)
-        (filterAttrs
-          (k: v: v.apps != null)
-          config.allSystems
-        );
+      mkIfNonEmptySet
+        (mapAttrs
+          (k: v: v.apps)
+          (filterAttrs
+            (k: v: v ? apps)
+            config.allSystems
+          ));
```

Comment:
This technique makes attributes such as `devShell.x86_64-linux.default` depend on `checks.aarch64-darwin`. (To pick a highly unrelated example)

You can see this when you define

```nix
  systems = [ "x86_64-linux" "aarch64-darwin" ];
  perSystem = { system, ... }: {
    checks = if system == "aarch64-darwin" then throw "not ok" else {};
  }
```

and then instantiate `devShell.x86_64-linux.default`.

Usually `checks.<system>` isn't too slow or fragile, but that's not an assumption we can make. Also strictness can lead to infinite recursions, which are horrible.